ludwig/hyperopt/execution.py

```diff
@@ -447,10 +447,17 @@ def _run_experiment(
         trial_id = tune.get_trial_id()
         trial_dir = Path(tune.get_trial_dir())
 
+        # Write out the unmerged config to the trial's local directory.
+        with open(os.path.join(trial_dir, "user_hyperparameters.json"), "w") as f:
+            json.dump(hyperopt_dict["config"], f)
+
```

Comment:
@jeffkinnison Would it make sense to potentially rename this to `trial_hyperparameters.json`?

Reply:
Also, would it make more sense to do this with `modified_config` (pre-merge with defaults, post merge with sampled values) as supposed to just `hyperopt_dict["config"]`? I'm assuming we also want to capture the values that were actually sampled for this trial, which I believe is stored in config and then merged into `hyperopt_dict["config"]` in the substitute_parameters call below. Let me know if I'm missing something!

Reply:
Sure, I can do that.

Reply:
That was a good call, and an oversight on my part, thanks!